lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -0,0 +1,72 @@
+const stream = require('stream');
+
+class ListRecordStream extends stream.Transform {
+    /**
+     * @constructor
+     * @param {Logger} logger logger
+     */
+    constructor(logger) {
+        super({ objectMode: true });
+        this._logger = logger;
+    }
+
+    /**
+     * Maps change stream operation type to
+     * backbeat supported types
+     * @param {string} operationType mongo opetation type
+     * @returns {string} supported operation type
+     */
+    _getType(operationType) {
+        switch (operationType) {
+            case 'insert':
+            case 'update':
+            case 'replace':
+                return 'put';
+            case 'delete':
+                return 'delete';
+            default:
+                this._logger.warn('Got unsupported operation', {
+                    method: 'ListRecordStream._getType',
+                    operationType,
+                });
+                return 'none';
+        }
+    }
+
+    /**
+     * parses json strings
+     * @param {string} blob sringified json
+     * @returns {Object} parsed json
+     */
+    _jsonParser(blob) {
+        let parsed = JSON.parse(blob);
+        if (typeof parsed === 'string') {
+            parsed = this._jsonParser(parsed);
+        }
```

Comment:
why do we need to (sometimes?) parse Json twice?
this should be predictable, and we should now what to expect in all cases... and adding some explanation in the code what/why happens would probably help readability

Reply:
the Kafka messages we consume contain over-stringified ChangeStreamDocuments, that's why they need to be parsed twice (it is always the case btw)